.github/workflows/rust.yml

```
@@ -21,4 +21,6 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+    - run: cargo fmt
+    - run: git diff --exit-code
```

```suggestion
    - run: cargo fmt --all -- --check
```

Should be enough to print the diff & exit with a non-zero code.